### ISSUE TITLE
Lightsail Commands Hanging

### ISSUE THREAD

**Issue by steinbachr** [boto/botocore#1183]
Attempting to perform any commands against lightsail, I'm getting an infinite hang. I've attempted the commands `aws lightsail get-static-ips`, `aws lightsail get-instances`, and `aws lightsail get-regions`  and all hang. 

Additionally, hang occurs when executing these commands from a python program using boto3 (pointing to an issue with *botocore* as opposed to an *awscli* or *boto3* issue). Finally, I've verified the issue is not with my internet connection / an intermittent AWS connection issue by successfully listing my s3 buckets with `aws s3 ls`

**Comment by dstufft** [comment]
Can you add the ``--debug`` flag to the ``aws`` command and show me the logs? Make sure you clean out any references to items you don't want made public.

**Comment by steinbachr** [comment]
see below:

```
(venv)~/D/P/Scratch ❯❯❯ aws lightsail get-regions --debug
2017-04-25 23:33:21,139 - MainThread - awscli.clidriver - DEBUG - CLI version: aws-cli/1.11.24 Python/2.7.9 Darwin/16.0.0 botocore/1.4.81
2017-04-25 23:33:21,140 - MainThread - awscli.clidriver - DEBUG - Arguments entered to CLI: ['lightsail', 'get-regions', '--debug']
2017-04-25 23:33:21,140 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function add_scalar_parsers at 0x108198500>
2017-04-25 23:33:21,140 - MainThread - botocore.hooks - DEBUG - Event session-initialized: calling handler <function inject_assume_role_provider_cache at 0x107efd0c8>
2017-04-25 23:33:21,141 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /usr/local/lib/python2.7/site-packages/botocore/data/lightsail/2016-11-28/service-2.json
2017-04-25 23:33:21,153 - MainThread - botocore.hooks - DEBUG - Event service-data-loaded.lightsail: calling handler <function register_retries_for_service at 0x107af1de8>
2017-04-25 23:33:21,153 - MainThread - botocore.handlers - DEBUG - Registering retry handlers for service: lightsail
2017-04-25 23:33:21,155 - MainThread - botocore.hooks - DEBUG - Event building-command-table.lightsail: calling handler <function add_waiters at 0x1081a08c0>
2017-04-25 23:33:21,160 - MainThread - awscli.clidriver - DEBUG - OrderedDict([(u'include-availability-zones', <awscli.arguments.BooleanArgument object at 0x1086200d0>), (u'no-include-availability-zones', <awscli.arguments.BooleanArgument object at 0x108620150>)])
2017-04-25 23:33:21,160 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.lightsail.get-regions: calling handler <function add_streaming_output_arg at 0x108198b18>
2017-04-25 23:33:21,160 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.lightsail.get-regions: calling handler <function add_cli_input_json at 0x107efdf50>
2017-04-25 23:33:21,160 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.lightsail.get-regions: calling handler <function unify_paging_params at 0x1081296e0>
2017-04-25 23:33:21,163 - MainThread - botocore.hooks - DEBUG - Event building-argument-table.lightsail.get-regions: calling handler <function add_generate_skeleton at 0x108113aa0>
2017-04-25 23:33:21,164 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.lightsail.get-regions: calling handler <bound method CliInputJSONArgument.override_required_args of <awscli.customizations.cliinputjson.CliInputJSONArgument object at 0x108620190>>
2017-04-25 23:33:21,164 - MainThread - botocore.hooks - DEBUG - Event before-building-argument-table-parser.lightsail.get-regions: calling handler <bound method GenerateCliSkeletonArgument.override_required_args of <awscli.customizations.generatecliskeleton.GenerateCliSkeletonArgument object at 0x108616ed0>>
2017-04-25 23:33:21,164 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.lightsail.get-regions.include-availability-zones: calling handler <function uri_param at 0x107d990c8>
2017-04-25 23:33:21,164 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.lightsail.get-regions.cli-input-json: calling handler <function uri_param at 0x107d990c8>
2017-04-25 23:33:21,164 - MainThread - botocore.hooks - DEBUG - Event load-cli-arg.lightsail.get-regions.generate-cli-skeleton: calling handler <function uri_param at 0x107d990c8>
2017-04-25 23:33:21,165 - MainThread - botocore.hooks - DEBUG - Event calling-command.lightsail.get-regions: calling handler <bound method GenerateCliSkeletonArgument.generate_json_skeleton of <awscli.customizations.generatecliskeleton.GenerateCliSkeletonArgument object at 0x108616ed0>>
2017-04-25 23:33:21,165 - MainThread - botocore.hooks - DEBUG - Event calling-command.lightsail.get-regions: calling handler <bound method CliInputJSONArgument.add_to_call_parameters of <awscli.customizations.cliinputjson.CliInputJSONArgument object at 0x108620190>>
2017-04-25 23:33:21,165 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: env
2017-04-25 23:33:21,165 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: assume-role
2017-04-25 23:33:21,165 - MainThread - botocore.credentials - DEBUG - Looking for credentials via: shared-credentials-file
2017-04-25 23:33:21,165 - MainThread - botocore.credentials - INFO - Found credentials in shared credentials file: ~/.aws/credentials
2017-04-25 23:33:21,166 - MainThread - botocore.loaders - DEBUG - Loading JSON file: /usr/local/lib/python2.7/site-packages/botocore/data/endpoints.json
2017-04-25 23:33:21,175 - MainThread - botocore.client - DEBUG - Registering retry handlers for service: lightsail
2017-04-25 23:33:21,176 - MainThread - botocore.hooks - DEBUG - Event creating-client-class.lightsail: calling handler <function add_generate_presigned_url at 0x107ac02a8>
2017-04-25 23:33:21,177 - MainThread - botocore.regions - DEBUG - Creating a regex based endpoint for lightsail, us-west-2
2017-04-25 23:33:21,177 - MainThread - botocore.args - DEBUG - The s3 config key is not a dictionary type, ignoring its value of: None
2017-04-25 23:33:21,212 - MainThread - botocore.endpoint - DEBUG - Setting lightsail timeout as (60, 60)
2017-04-25 23:33:21,212 - MainThread - botocore.hooks - DEBUG - Event before-parameter-build.lightsail.GetRegions: calling handler <function generate_idempotent_uuid at 0x107af1848>
2017-04-25 23:33:21,212 - MainThread - botocore.endpoint - DEBUG - Making request for OperationModel(name=GetRegions) (verify_ssl=True) with params: {'body': '{}', 'url': u'https://lightsail.us-west-2.amazonaws.com/', 'headers': {'User-Agent': 'aws-cli/1.11.24 Python/2.7.9 Darwin/16.0.0 botocore/1.4.81', 'Content-Type': u'application/x-amz-json-1.1', 'X-Amz-Target': u'Lightsail_20161128.GetRegions'}, 'context': {'client_region': 'us-west-2', 'has_streaming_input': False, 'client_config': <botocore.config.Config object at 0x1087549d0>}, 'query_string': '', 'url_path': '/', 'method': u'POST'}
2017-04-25 23:33:21,213 - MainThread - botocore.hooks - DEBUG - Event request-created.lightsail.GetRegions: calling handler <bound method RequestSigner.handler of <botocore.signers.RequestSigner object at 0x108754950>>
2017-04-25 23:33:21,213 - MainThread - botocore.auth - DEBUG - Calculating signature using v4 auth.
2017-04-25 23:33:21,213 - MainThread - botocore.auth - DEBUG - CanonicalRequest:
POST
/

content-type:application/x-amz-json-1.1
host:lightsail.us-west-2.amazonaws.com
x-amz-date:20170426T033321Z
x-amz-target:Lightsail_20161128.GetRegions

content-type;host;x-amz-date;x-amz-target
44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a
2017-04-25 23:33:21,214 - MainThread - botocore.auth - DEBUG - StringToSign:
AWS4-HMAC-SHA256
20170426T033321Z
20170426/us-west-2/lightsail/aws4_request
5a222eed7894dad3470bf8dd20b7f53b435fe35f81827b6a88f8f1224b23924a
2017-04-25 23:33:21,214 - MainThread - botocore.auth - DEBUG - Signature:
c14db5f7d6cb61459933a5ea202a8deede269976585d109ff7edc11f6248e083
2017-04-25 23:33:21,218 - MainThread - botocore.endpoint - DEBUG - Sending http request: <PreparedRequest [POST]>
2017-04-25 23:33:21,219 - MainThread - botocore.vendored.requests.packages.urllib3.connectionpool - INFO - Starting new HTTPS connection (1): lightsail.us-west-2.amazonaws.com
```

**Comment by steinbachr** [comment]
ok, I was actually able to resolve the issue myself. The issues were two-fold:

1) Your default region has to be set to `us-east-1`. Mine was set to `us-west-1`. 
2) After setting default region, I started getting a different error that my IAM user didn't have permission to access lightsail resources. So, I created a Policy granting access to `lightsail:*` and attached it to my user. However, I was still getting permission denied errors. It was only by explicitly setting context on the resource (i.e. `lightsail:GetBlueprints`) that it worked.
